### PR TITLE
Small search dialog optimizations

### PR DESCRIPTION
**Pull request by jpstotz**
When opening the search dialog for the first time the layout was a bit strange, it seems the WrapLayout was reserving space for placing the two search option boxes stacked vertically, but later redecided and placed them horizontally but keeping the vertical space. 
This was may be related to the fact that the target window size was restored after creating all components and additionally an fixed intermediate window size was set. Seems that this confused the Java layouting engines.

Additionally I modified the search dialog to show a red background color in case an invalid regex was entered.

### CODE REVIEW

File: jadx-gui/src/main/java/jadx/gui/ui/SearchDialog.java [hunk 1]
```diff
@@ -246,8 +248,6 @@ public void keyReleased(KeyEvent e) {
 		});
 
 		setTitle(NLS.str("menu.text_search"));
-		pack();
-		setSize(800, 500);
```
Comment (skylot) [inline jadx-gui/src/main/java/jadx/gui/ui/SearchDialog.java:250]: Hm, if you don't call `pack()` or `setSize` first run will show zero size window.
So I suggest to check if window pos not applied and call `pack()`: 
```java
if (!mainWindow.getSettings().loadWindowPos(this)) {
		pack();
}
```
Also, window pos is ignored for some cases: like moving window outside of screen bounds (check `JadxSettings#loadWindowPos()` code and PR #521).

**Review by skylot — CHANGES_REQUESTED** [review]

File: jadx-gui/src/main/java/jadx/gui/ui/SearchDialog.java (hunk 1 above)
Comment (jpstotz) [reply jadx-gui/src/main/java/jadx/gui/ui/SearchDialog.java:250]: You are correct, I did not consider the case when Jadx is started the first time. I will make some more tests. 

Unfortunately your proposed changes doesn't fix all problems. I assume that there is a bug in the calculation of WrapLayout. I will check that tomorrow.

Comment (jpstotz) [reply jadx-gui/src/main/java/jadx/gui/ui/SearchDialog.java:250]: It took a little bit longer to check the WrapLayout. I updated the code so the dialog size is now set if the dialog position can not be restored from settings.

**Comment by jpstotz** [comment]
The major change is in WrapLayout the change of the comparison:

* old: `rowWidth + width > maxWidth`
* new: `rowWidth + width >= maxWidth`

Based on my tests the greater-euqal comparison is required to make the WrapLayout work correctly.

**Comment by jpstotz** [comment]
I am sorry - sometimes I really don't know what the hell git respectively IntelliJ is doing. I never configured this repo as remote for that branch. 
Now the the push was done to the correct repo.

**Review by skylot — APPROVED** [review]

**Comment by skylot** [comment]
@jpstotz thanks!

**Pull request merged**